content/custom_resources/glossary.md

```diff
@@ -0,0 +1,607 @@
+load_current_value do |new_resource|
+  if ::File.exist?(new_resource.path)
+    content IO.read(new_resource.path)
+    mode ::File.stat(new_resource.path).mode
+  end
+end
+
+action :create do
+
+  # If the value of content has changed
+  # write file
+  converge_if_changed :content do
+    IO.write(new_resource.path, new_resource.content)
+  end
+
+  # If the value of mode has changed then
+  # chmod file
+  converge_if_changed :mode do
+    ::File.chmod(new_resource.mode, new_resource.path)
+  end
+end
+```
+
+Chef Infra Client will only update the property values that require
+updates and will not make changes when the property values are already
+in the desired state.
+
+## default_action
+
+The default action in a custom resource is, by default, the first action
+listed in the custom resource. For example, action `aaaaa` is the
+default resource:
+
+```ruby
+property :property_name, RubyType, default: 'value'
+
+...
+
+action :aaaaa do
+ # the first action listed in the custom resource
+end
+
+action :bbbbb do
+ # the second action listed in the custom resource
+end
+```
+
+The `default_action` method may also be used to specify the default
+action. For example:
+
+```ruby
+property :property_name, RubyType, default: 'value'
+
+# Define bbbbb aas the default action
+default_action :bbbbb
+
+action :aaaaa do
+ # the first action listed in the custom resource
+end
+
+action :bbbbb do
+ # the second action listed in the custom resource
+end
+```
+
+## load_current_value
+
+Use the `load_current_value` method to load the specified property
+values from the node, and then use those values when the resource is
+converged. This method may take a block argument.
+
+```ruby
+property :path, String
+property :content, String
+property :mode, String
+
+load_current_value do |new_resource|
+  if ::File.exist?(new_resource.path)
+    content IO.read(new_resource.path)
+    mode ::File.stat(new_resource.path).mode
+  end
+end
+```
+
+Use the `load_current_value` method to guard against property value being replaced. For example:
+
+```ruby
+property :homepage, String
+property :page_not_found, String
+
+load_current_value do
+  if ::File.exist?('/var/www/html/index.html')
+    homepage IO.read('/var/www/html/index.html')
+  end
+
+  if ::File.exist?('/var/www/html/404.html')
+    page_not_found IO.read('/var/www/html/404.html')
+  end
+end
+```
+
+This ensures the values for `homepage` and `page_not_found` are not
+changed to the default values when Chef Infra Client configures the
+node.
+
+## current_value_does_not_exist!
+
+When using the `load_current_value` block, use `current_value_does_not_exist!` to indicate that the value does not exist and that `current_resource` should therefore be `nil`.
+
+```ruby
+load_current_value do
+  current_value_does_not_exist!
+end
+```
+
+## new_resource.property
+
+Custom resources are designed to use core resources that are built into Chef. In some cases, it may be necessary to specify a property in the
```

Comment:
```suggestion
Custom resources are designed to use core resources that are built into Chef Infra. In some cases, it may be necessary to specify a property in the
```